src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSparkEngine.java

```diff
@@ -31,29 +28,45 @@
     private final JavaSparkContext ctx;
     private final String indexFileName;
     private final Broadcast<SAMFileHeader> broadcastHeader;
+    private final boolean pairedMode;
 
-    public BwaSparkEngine(final JavaSparkContext ctx,
-                          final String indexFileName,
-                          SAMFileHeader inputHeader,
-                          final SAMSequenceDictionary refDictionary) {
+    public BwaSparkEngine( final JavaSparkContext ctx,
+                           final String indexFileName,
+                           final SAMFileHeader inputHeader,
+                           final SAMSequenceDictionary refDictionary ) {
+        this(ctx, indexFileName, inputHeader, refDictionary, true);
+    }
+
+    public BwaSparkEngine( final JavaSparkContext ctx,
+                           final String indexFileName,
+                           SAMFileHeader inputHeader,
+                           final SAMSequenceDictionary refDictionary,
+                           final boolean pairedMode ) {
         Utils.nonNull(indexFileName);
         Utils.nonNull(inputHeader);
+        if ( pairedMode && inputHeader.getSortOrder() != SAMFileHeader.SortOrder.queryname ) {
+            throw new GATKException("Input must be queryname sorted to use paired alignment mode.");
```

Comment:
I think this should be a `UserException` since the user provided a bad bam file.